src/main/java/strategies/impl/AllStrategyImpl.java

```diff
@@ -0,0 +1,29 @@
+package strategies.impl;
+
+import strategies.SearchStrategy;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static util.QueryFilter.queryFilter;
+
+public class AllStrategyImpl implements SearchStrategy {
+
+    @Override
+    public List<String> search(List<String> stringsList, Map<String, List<Integer>> searchMap, String query) {
+        String[] words = queryFilter(query);
+        List<String> found = new ArrayList<>();
+
+        searchMap.getOrDefault(words[0], new ArrayList<>())
+                .forEach(index -> {
+                    boolean temporaryIndex = Arrays.stream(words)
+                            .anyMatch(word -> !stringsList.get(index).toUpperCase().contains(word));
+                    if (!temporaryIndex) {
+                        found.add(stringsList.get(index));
+                    }
+                });
+        return found;
```

Comment:
It looks the same as before. Try to avoid creating collections that store the state (in current case `List<String> found`) of the stream in most cases you should get data from the Stream, that's why you can see different terminated operations that return the results.